pkg/cloudprovider/aws/subnets.go

```diff
@@ -41,20 +41,27 @@ func NewSubnetProvider(ec2api ec2iface.EC2API) *SubnetProvider {
 }
 
 func (s *SubnetProvider) Get(ctx context.Context, constraints *v1alpha1.Constraints) ([]*ec2.Subnet, error) {
-	// Get subnets
-	subnets, err := s.getSubnets(ctx, s.getFilters(constraints))
+	filters := getFilters(constraints)
+	hash, err := hashstructure.Hash(filters, hashstructure.FormatV2, nil)
 	if err != nil {
 		return nil, err
 	}
-	// Fail if no subnets found
-	if len(subnets) == 0 {
-		return nil, fmt.Errorf("no subnets exist given constraints")
+	if subnets, ok := s.cache.Get(fmt.Sprint(hash)); ok {
+		return subnets.([]*ec2.Subnet), nil
+	}
+	output, err := s.ec2api.DescribeSubnetsWithContext(ctx, &ec2.DescribeSubnetsInput{Filters: filters})
+	if err != nil {
+		return nil, fmt.Errorf("describing subnets %s, %w", pretty.Concise(filters), err)
+	}
+	if len(output.Subnets) == 0 {
+		return nil, fmt.Errorf("no subnets matched selector %v", constraints.SubnetSelector)
 	}
-	// Return subnets
-	return subnets, nil
+	s.cache.Set(fmt.Sprint(hash), output.Subnets, CacheTTL)
```

Comment:
IIUC there's a SetDefault method you can use to avoid setting CacheTTL

Reply:
Ok, I'll clean that up in another PR - looks like a bunch o' cases.